doc.py

```diff
@@ -161,16 +196,49 @@ def extract_text(self) -> str:
         return "\n\n".join(paragraphs)
 
 
+class ODTDocument(Document):
+    """ OpenDocument format. """
+
+    def extract_text(self) -> str:
+        textdoc = load_odf(BytesIO(self.data))
+        paragraphs = textdoc.getElementsByType(odf_text.P)  # Find all paragraphs
+        ptexts = [teletype.extractText(p) for p in paragraphs]
+        return "\n\n".join(ptexts)
+
+
 # Map file mime type to document class
 MIMETYPE_TO_DOC_CLASS: Dict[str, Type[Document]] = {
     "text/plain": PlainTextDocument,
     "text/html": HTMLDocument,
     "text/rtf": RTFDocument,
+    "application/pdf": PDFDocument,
+    "application/x-pdf": PDFDocument,
     "application/rtf": RTFDocument,
-    # "application/pdf": PDFDocument,
-    # "application/x-pdf": PDFDocument,
+    "application/vnd.oasis.opendocument.text": ODTDocument,
     # Yes, really!
     "application/vnd.openxmlformats-officedocument.wordprocessingml.document": DocxDocument,
 }
 
 SUPPORTED_DOC_MIMETYPES = frozenset(MIMETYPE_TO_DOC_CLASS.keys())
+
+
+def doc_class_for_mime_type(mime_type: str) -> Type[Document]:
+    assert mime_type in SUPPORTED_DOC_MIMETYPES
+    return MIMETYPE_TO_DOC_CLASS[mime_type]
+
+
+SUFFIX_TO_DOC_CLASS: Dict[str, Type[Document]] = {
```

Comment:
Strictly speaking, immutable/read-only Dicts such as these should be typed as a `Mapping`, not `Dict`